website/docs/attached-class.md

```diff
@@ -91,3 +91,69 @@ T.reveal_type(Child.make) # Child
 
 Child.make.say_hi # No error now, as Child.make has the type Child
 ```
+
+## `T.attached_class` as an argument?
+
+At this point, it's natural to think about writing methods that accept arguments
+that take values of type `T.attached_class`; you would get more specific type
+information about the arguments when calling from a sub-class, so why not?
+Consider this example:
+
+```ruby
+class Parent
+  extend T::Sig
+
+  sig {returns(T.attached_class)}
+  def self.make
+    new
+  end
+
+  sig {params(x: T.attached_class).void}
```

Comment:
Can we add an inline comment to this snippet saying that they should **not** expect this line to work? It makes it easier for people who are not reading and instead just skimming.